### PR TITLE
Progress #1232 -- Replacing `UpdateStats` with `OnReplication`

### PR DESCRIPTION
**Pull request by Killfrra**

### CODE REVIEW

File: PacketDefinitions420/PacketNotifier.cs [hunk 1]
```diff
@@ -3736,15 +3736,20 @@ public void NotifyUpdatedStats(IAttackableUnit u, int userId = 0, bool partial =
         {
```
Comment (cabeca1143) [inline PacketDefinitions420/PacketNotifier.cs]: Update the function's name to `NotifyOnReplication` and rearrange it in alphabetical order

File: PacketDefinitions420/PacketNotifier.cs [hunk 2]
```diff
@@ -3811,7 +3816,12 @@ void NotifyEnterTeamVision(IGameObject obj, TeamId team, int userId = 0, GamePac
                 }
                 var healthbarPacket = ConstructEnterLocalVisibilityClientPacket(obj);
                 //TODO: try to include it to packets too?
-                var us = new UpdateStats(u.Replication, false);
+                var us = new OnReplication(){
```
Comment (cabeca1143) [inline PacketDefinitions420/PacketNotifier.cs:3917]: Instead of calling `new OnReplication` here, why not use the replication function itself?

Comment (Killfrra) [reply PacketDefinitions420/PacketNotifier.cs:3917]: I don't quite understand. I can add another function like `ConstructOnReplication(IGameObject for, bool partial)`. But you will still have to get rid of it later, when the transition is made from creating a packet for each unit to creating one packet for all units.

File: PacketDefinitions420/PacketNotifier.cs (hunk 1 above)
Comment (Killfrra) [reply PacketDefinitions420/PacketNotifier.cs]: :heavy_check_mark:

**Review by rarfiles2 — CHANGES_REQUESTED** [review]
add a pcicutre of karsadin

File: PacketDefinitions420/PacketNotifier.cs (hunk 2 above)
Comment (rarfiles2) [reply PacketDefinitions420/PacketNotifier.cs:3916]: How do you not understand this? I have known this for ages, you piece of shit.

**Review by lzardy — APPROVED** [review]
All good.

**Pull request merged**